project/webapp/model/productModel.js

```diff
@@ -1,20 +1,22 @@
 sap.ui.define([
-    "sap/ui/model/json/JSONModel",
-], 
-    /**
-     * provide app-view type models (as in the first "V" in MVVC)
-     * 
-     * @param {typeof sap.ui.model.json.JSONModel} JSONModel
-     * 
-     * @returns {Function} createModel() for providing runtime products mock data
-     */
-    function (JSONModel) {
-        "use strict";
+  "sap/ui/model/json/JSONModel",
+],
+  /**
+   * provide app-view type models (as in the first "V" in MVVC)
+   * 
+   * @param {typeof sap.ui.model.json.JSONModel} JSONModel
+   * 
+   * @returns {Function} getModel() for providing runtime products mock data
+   */
+  function (JSONModel) {
+    "use strict";
 
-        return {
-            createModel: function () {
-                const oModel = new JSONModel();
-                return oModel;
-        }
+    return {
+      getModel: function () {
```

Comment:
method getModel should only return json model instance, to create that model we can add other method like 'init', which will create and load data to your json model
init method should be called from controller
always try to follow principle: method should do only these actions he was invented for. try to avoid adding unexpected extra logic